### PR TITLE
C++17

### PR DESCRIPTION
**Pull request by Nekotekina**
This is not intended to be a fully complete PR, it just needs to compile and work.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -41,6 +44,12 @@ install:
       wget https://mirrors.kernel.org/ubuntu/pool/universe/v/vulkan/libvulkan-dev_1.1.73+dfsg-1_amd64.deb;
       sudo apt-get install --only-upgrade dpkg;
       sudo dpkg -i libglew2.0_2.0.0-5_amd64.deb libglew-dev_2.0.0-5_amd64.deb libvulkan1_1.1.73+dfsg-1_amd64.deb libvulkan-dev_1.1.73+dfsg-1_amd64.deb;
+      if [ "$CXX" = "clang++-6.0" ]; then
+        wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -;
+        sudo apt-add-repository "deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty-6.0 main";
+        sudo apt-get update;
+        sudo apt-get install -y clang-6.0 lld-6.0;
```
Comment (hcorion) [inline .travis.yml]: Just add clang-6.0 and clang++-6.0 in the packages section, you shouldn't need lld, and rpcs3 doesn't link with lld 6.0 anyway. and add `llvm-toolchain-trusty-6.0` to the *sources* section, right before the *packages* section.

**Comment by hcorion** [comment]
Also need to bump the README.md minimum clang and gcc versions.

File: .gitmodules [hunk 2]
```diff
@@ -23,7 +23,7 @@
 	url = https://github.com/madler/zlib
 [submodule "3rdparty/hidapi"]
 	path = 3rdparty/hidapi
-	url = https://github.com/RPCS3/hidapi
+	url = https://github.com/Zangetsu38/hidapi
```
Comment (AniLeo) [inline .gitmodules]: Don't forget to point this bacc to our fork

**Comment by AniLeo** [comment]
Need to update here too
https://github.com/RPCS3/rpcs3/blob/master/llvm_build/UpdateProjectFiles.bat#L4

File: README.md [hunk 3]
```diff
@@ -32,7 +32,7 @@ If you want to contribute please take a look at the [Coding Style](https://githu
 
 ### Linux
 * [Qt 5.10+](https://www.qt.io/download-open-source/)
-* GCC 5.1+ or Clang 3.5.0+ ([not GCC 6.1](https://github.com/RPCS3/rpcs3/issues/1691))
+* GCC 7.3+ or Clang 5.0+ ([not GCC 6.1](https://github.com/RPCS3/rpcs3/issues/1691))
```
Comment (AniLeo) [inline README.md]: You can remove "not GCC 6.1" since we require GCC 7+ now

**Pull request merged**